torch/_inductor/codecache.py

```diff
@@ -808,6 +808,11 @@ def __init__(
         # Alignment checks
         self.inputs_to_check = inputs_to_check
 
+        # This device index is usually already encoded by the device of the inputs
+        # but fx graphs don't necessarily have tensor inputs
+        if torch.cuda.is_available():
+            self.default_cuda_device_index = torch.cuda.current_device()
```

Comment:
partially a PSA / question for @albanD  - to what extent do you expect people to be using `torch.accelerator.current_device()` + PT2 today? James and I talked about it offline a bit - we'll need to do something similar for other accelerators in the long run to avoid accelerator-specific warm cache problems

Reply:
Yes you should use torch.accelerator instead of cuda for these things. They lead to the same thing for the cuda device but you also get rocm/mtia/xpu support for free.

Reply:
Is it guaranteed that:

if torch.cuda.is_available(), then `torch.accelerator.is_available()`? 
And that `torch.cuda.current_device() == torch.accelerator.current_device`? 

If so happy to change

Reply:
Seems like it works in my tests, will run with it!